syntax = "proto2";

message Reader {
  oneof reader_oneof {
    CaptionGraphReader caption_graph_reader = 5;
  }
}

message CaptionGraphReader {
  // Pattern of the input files.
  repeated string input_pattern = 1;

  // Shuffle buffer size.
  optional int32 shuffle_buffer_size = 2 [default = 1000];

  // Interleave cycle length.
  optional int32 interleave_cycle_length = 3 [default = 1];

  // Number of parallel calls.
  optional int32 num_parallel_calls = 5 [default = 5];

  // Batch size.
  optional int32 batch_size = 6 [default = 16];

  // Prefetch buffer size.
  optional int32 prefetch_buffer_size = 7 [default = 200];

  // Dimensions of the pre-extracted FRCNN features.
  optional int32 feature_dimensions = 8 [default = 1536];

  // Maximum number of proposals.
  optional int32 max_n_proposal = 9 [default = 300];

  // For DEBUG purpose, use a fixed graph instead of sample one.
  optional int32 graph_index = 10 [default =-1];
}
